From: Debian Games Team <pkg-games-devel@lists.alioth.debian.org>
Date: Sun, 4 Jan 2015 20:49:22 +0100
Subject: debianization

---
 cmake/Modules/FindIrrlicht.cmake | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cmake/Modules/FindIrrlicht.cmake b/cmake/Modules/FindIrrlicht.cmake
index bd00422..f94159a 100644
--- a/cmake/Modules/FindIrrlicht.cmake
+++ b/cmake/Modules/FindIrrlicht.cmake
@@ -50,7 +50,7 @@ else()
 		/usr/include/irrlicht
 	)
 
-	FIND_LIBRARY(IRRLICHT_LIBRARY NAMES libIrrlicht.a Irrlicht
+	FIND_LIBRARY(IRRLICHT_LIBRARY NAMES libIrrlicht.so
 		PATHS
 		/usr/local/lib
 		/usr/lib
