From: Markus Koschany <apo@gambaru.de>
Date: Sun, 4 Jan 2015 22:57:24 +0100
Subject: desktop file

Forwarded: https://github.com/minetest/minetest/pull/2165
---
 misc/minetest.desktop | 1 +
 1 file changed, 1 insertion(+)

diff --git a/misc/minetest.desktop b/misc/minetest.desktop
index 3f5b9c4..e5ce24a 100644
--- a/misc/minetest.desktop
+++ b/misc/minetest.desktop
@@ -8,4 +8,5 @@ Terminal=false
 Type=Application
 Categories=Game;
 StartupNotify=false
+Keywords=sandbox;world;mining;crafting;blocks;nodes;multiplayer;roleplaying;
 
