From: Markus Koschany <apo@gambaru.de>
Date: Sat, 21 Feb 2015 16:45:20 +0100
Subject: FindIrrlicht.cmake

Forwarded: https://github.com/minetest/minetest/pull/2371
---
 cmake/Modules/FindIrrlicht.cmake | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cmake/Modules/FindIrrlicht.cmake b/cmake/Modules/FindIrrlicht.cmake
index dce78fb..a84765d 100644
--- a/cmake/Modules/FindIrrlicht.cmake
+++ b/cmake/Modules/FindIrrlicht.cmake
@@ -44,7 +44,7 @@ else()
 		/usr/include/irrlicht
 	)
 
-	FIND_LIBRARY(IRRLICHT_LIBRARY NAMES libIrrlicht.a Irrlicht
+	FIND_LIBRARY(IRRLICHT_LIBRARY NAMES libIrrlicht.so libIrrlicht.a Irrlicht
 		PATHS
 		/usr/local/lib
 		/usr/lib
