From: Debian Games Team <pkg-games-devel@lists.alioth.debian.org>
Date: Sun, 4 Jan 2015 20:49:22 +0100
Subject: gcc47

---
 src/filesys.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/filesys.cpp b/src/filesys.cpp
index 7847156..e5fec32 100644
--- a/src/filesys.cpp
+++ b/src/filesys.cpp
@@ -193,6 +193,7 @@ std::string TempPath()
 
 #else // POSIX
 
+#include <unistd.h>
 #include <sys/types.h>
 #include <dirent.h>
 #include <sys/stat.h>
