From: Markus Koschany <apo@debian.org>
Date: Sat, 26 Aug 2017 23:48:05 +0200
Subject: luajit 2.1

---
 cmake/Modules/FindLuaJIT.cmake | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cmake/Modules/FindLuaJIT.cmake b/cmake/Modules/FindLuaJIT.cmake
index e4335d8..dd6fd04 100644
--- a/cmake/Modules/FindLuaJIT.cmake
+++ b/cmake/Modules/FindLuaJIT.cmake
@@ -9,7 +9,7 @@
 FIND_PATH(LUA_INCLUDE_DIR luajit.h
 	HINTS
 	$ENV{LUA_DIR}
-	PATH_SUFFIXES include/luajit-2.0 include/luajit-5_1-2.0 include
+	PATH_SUFFIXES include/luajit-2.1 include/luajit-5_1-2.0 include
 	PATHS
 	~/Library/Frameworks
 	/Library/Frameworks
