From: Markus Koschany <apo@debian.org>
Date: Fri, 20 Feb 2015 22:46:25 +0100
Subject: rawlua

---
 src/CMakeLists.txt |    7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

Index: b/src/CMakeLists.txt
===================================================================
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -167,10 +167,9 @@ else()
 	message (STATUS "LuaJIT detection disabled! (ENABLE_LUAJIT=0)")
 endif()
 if(NOT USE_LUAJIT)
-	message(STATUS "LuaJIT not found, using bundled Lua.")
-	set(LUA_LIBRARY "lua")
-	set(LUA_INCLUDE_DIR "${PROJECT_SOURCE_DIR}/lua/src")
-	add_subdirectory(lua)
+	message (STATUS "LuaJIT not found, using the version of Lua provided by the system.")
+	set(LUA_INCLUDE_DIR "/usr/include/lua5.1/")
+	set(LUA_LIBRARY "lua5.1")
 endif()
 
 find_package(GMP REQUIRED)
