From: Debian Games Team <pkg-games-devel@lists.alioth.debian.org>
Date: Sun, 4 Jan 2015 20:49:23 +0100
Subject: rawlua

---
 src/CMakeLists.txt | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index b308ad2..37dd3f9 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -261,9 +261,9 @@ if(LUA_LIBRARY AND LUA_INCLUDE_DIR)
 endif (LUA_LIBRARY AND LUA_INCLUDE_DIR)
 
 if(NOT USE_LUAJIT)
-	message (STATUS "LuaJIT not found, using bundled Lua.")
-	set(LUA_INCLUDE_DIR "${PROJECT_SOURCE_DIR}/lua/src")
-	set(LUA_LIBRARY "lua")
+	message (STATUS "LuaJIT not found, using the version of Lua provided by the system.")
+	set(LUA_INCLUDE_DIR "/usr/include/lua5.1/")
+	set(LUA_LIBRARY "lua5.1")
 endif(NOT USE_LUAJIT)
 
 mark_as_advanced(LUA_LIBRARY)
